packages/ra-ui-materialui/src/input/AutocompleteInput.tsx

```diff
@@ -208,48 +78,34 @@ export const AutocompleteInput = (props: AutocompleteInputProps) => {
         ...rest,
     });
 
-    const [filterValue, setFilterValue] = useState('');
-
-    const getSuggestionFromValue = useCallback(
-        value => choices.find(choice => get(choice, optionValue) === value),
-        [choices, optionValue]
-    );
-
     const selectedItem = useMemo(
-        () => getSuggestionFromValue(input.value) || null,
-        [input.value, getSuggestionFromValue]
+        () =>
+            choices?.find(choice => get(choice, optionValue) === input.value) ||
+            null,
+        [choices, input.value, optionValue]
     );
 
-    const doesQueryMatchSuggestion = useMemo(() => {
+    useEffect(() => {
         if (isValidElement(optionText)) {
```

Comment:
why put that in an effect?

Reply:
Because we rely on props and we want the warning to be displayed if needed when they change?